Gulpfile.js

```diff
@@ -265,6 +278,124 @@ gulp.task('test-client-travis', ['build'], function () {
 
 gulp.task('travis', [process.env.GULP_TASK || '']);
 
+//Documentation
+gulp.task('lint-docs', function () {
+    function lintFiles (files) {
+        return new Promise(function (resolve, reject) {
+            var config = {
+                'files':  files,
+                'config': require('./.markdownlint.json')
+            };
+
+            markdownlint(config, function (err, result) {
+                var lintErr = err || result && result.toString();
+
+                if (lintErr)
+                    reject(lintErr);
+                else
+                    resolve();
+            });
+        });
+    }
+
+    return globby('./docs/articles/**/*.md').then(lintFiles);
+});
+
+gulp.task('clear-site', ['lint-docs'], function (cb) {
+    del('./site/**', function (e) {
+        cb(e);
+    });
+});
+
+gulp.task('fetch-assets-repo', ['clear-site'], function (cb) {
+    git.clone('https://github.com/VasilyStrelyaev/testcafe-gh-page-assets.git', { args: './site' }, function (e) {
+        if (e) return cb(e);
+        cb();
+    });
+});
+
+gulp.task('put-in-articles', ['fetch-assets-repo'], function () {
+    return gulp.src('./docs/articles/**/*')
+		.pipe(gulp.dest('./site/src'));
+});
+
+gulp.task('put-in-navigation', ['put-in-articles'], function () {
+    return gulp.src('./docs/_data/**/*')
+		.pipe(gulp.dest('./site/src/_data'));
+});
+
+gulp.task('put-in-templates', ['put-in-navigation'], function () {
+    return gulp.src('./docs/_includes/**/*')
+		.pipe(gulp.dest('./site/src/_includes'));
+});
+
+gulp.task('put-in-docs', ['put-in-articles', 'put-in-navigation', 'put-in-templates']);
+
+gulp.task('prepare-website', ['fetch-assets-repo', 'put-in-docs']);
+
+gulp.task('build-website', ['prepare-website'], function (cb) {
+    var cp = spawn('jekyll.bat', ['build', '--source', './site/src/', '--destination', './site/deploy']);
```

Comment:
What is `jekyll.bat`? This is windows-specific. So, I will not be able to run docs site on my Mac